.travis.yml

```diff
@@ -9,48 +9,49 @@ services:
-      # Create kind cluster
-      - |
-        if [ $(kind get clusters) ]; then
-          export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
-        else
-          kind create cluster
-        fi
-      # Get kubeconfig
-      - export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
-      # Build image and tag as gatekeeper-e2e:latest
-      - make docker-build REPOSITORY=gatekeeper-e2e
-      # Load Gatekeeper image into kind cluster
-      - kind load docker-image --name kind gatekeeper-e2e:latest
-      # Download and install kubectl
-      - curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl && chmod +x ./kubectl && sudo mv kubectl /usr/local/bin/
-      # Download and install kustomize
-      - curl -L https://github.com/kubernetes-sigs/kustomize/releases/download/v${KUSTOMIZE_VERSION}/kustomize_${KUSTOMIZE_VERSION}_linux_amd64 --output kustomize && chmod +x kustomize && sudo mv kustomize /usr/local/bin/
+      - make e2e-bootstrap
+      - export KUBECONFIG="$(kind get kubeconfig-path --name="kind")" 
+      - make e2e-build-load-image IMG=gatekeeper-e2e:latest
+      - make patch-image IMG=gatekeeper-e2e:latest USE_LOCAL_IMG=true
       script:
-      - make deploy
+      - make deploy 
       - make test-e2e
       - echo -e '\n\n======= manager logs =======\n\n' && kubectl logs -n gatekeeper-system gatekeeper-controller-manager-0 manager
-    - stage: "build, test, deploy"
-      script: make
+    - stage: "dev deploy"
+      if: repo = open-policy-agent/gatekeeper AND type != pull_request
+      script: 
+      - make manager
       deploy:
         - provider: script
           skip_cleanup: true
           script: make travis-dev-deploy
           on:
             repo: open-policy-agent/gatekeeper
-            branch: master
+            branch: master # only push dev image tag for master commits
             tags: false
+    - stage: "release deploy, verify release"
+      if: repo = open-policy-agent/gatekeeper AND type != pull_request AND tag IS present
+      script: 
+      - make manager
+      before_deploy:
+        - make e2e-bootstrap
+        - export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
+      deploy:
         - provider: script
           skip_cleanup: true
-          script: make travis-dev-release
+          script: make travis-release-deploy IMG=quay.io/open-policy-agent/gatekeeper:${TRAVIS_TAG} VERSION=${TRAVIS_TAG}
           on:
             repo: open-policy-agent/gatekeeper
-            branch: master
+            branch: master # only push release image tag for tags on master
+            tags: true
+        - provider: script
+          skip_cleanup: true
+          script: make e2e-verify-release IMG=quay.io/open-policy-agent/gatekeeper:${TRAVIS_TAG} USE_LOCAL_IMG=false
```

Comment:
this step can be run as its own stage to guarantee we're verifying the release on a blank machine.